plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/MockStorage.java

```diff
@@ -356,8 +364,21 @@ public Blob compose(ComposeRequest composeRequest) {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public StorageBatch batch() {
-        return null;
+        final StorageBatch batch = mock(StorageBatch.class);
+        StorageBatchResult<Boolean> result = mock(StorageBatchResult.class);
+        doAnswer(answer -> {
+            BatchResult.Callback<Boolean, Exception> callback  = (BatchResult.Callback<Boolean, Exception>) answer.getArguments()[0];
+            callback.success(true);
+            return null;
+        }).when(result).notify(any(BatchResult.Callback.class));
+        when(batch.delete(any(BlobId.class), anyVararg())).thenAnswer(invocation -> {
```

Comment:
can we somehow check that no other method is called on this mock?

Reply:
Jup done :)